src/Microsoft.Diagnostics.EventFlow.Inputs.EventFlowTelemetryProcessor/EventFlowTelemetryProcessor.cs

```diff
@@ -0,0 +1,461 @@
+                    AddMetricValues(eventData, dependencyCall.Metrics);
+                    goto AddCommonProperties;
+                }
+
+                var metric = item as MetricTelemetry;
+                if (metric != null)
+                {
+                    eventData.Level = LogLevel.Informational;
+                    AddMetricProperties(eventPayload, metric);
+                    goto AddCommonProperties;
+                }
+
+                var exception = item as ExceptionTelemetry;
+                if (exception != null)
+                {
+                    if (exception.SeverityLevel.HasValue)
+                    {
+                        eventData.Level = ToLogLevel[(int)exception.SeverityLevel.Value];
+                    }
+                    else
+                    {
+                        eventData.Level = LogLevel.Warning;
+                    }
+
+                    AddExceptionProperties(eventPayload, exception);
+                    AddMetricValues(eventData, exception.Metrics);
+                    goto AddCommonProperties;
+                }
+
+                var pageView = item as PageViewTelemetry;
+                if (pageView != null)
+                {
+                    eventData.Level = LogLevel.Informational;
+                    AddPageViewProperties(eventPayload, pageView);
+                    AddMetricValues(eventData, pageView.Metrics);
+                    goto AddCommonProperties;
+                }
+
+                var availability = item as AvailabilityTelemetry;
+                if (availability != null)
+                {
+                    eventData.Level = availability.Success ? LogLevel.Informational : LogLevel.Warning;
+                    AddAvailabilityProperties(eventPayload, availability);
+                    AddMetricValues(eventData, availability.Metrics);
+                    goto AddCommonProperties;
+                }
+
+            AddCommonProperties:
+                var itemWithProperties = item as ISupportProperties;
+                if (itemWithProperties != null)
+                {
+                    foreach (var property in itemWithProperties.Properties)
+                    {
+                        AddPayloadProperty(eventData, property.Key, property.Value);
+                    }
+                }
+
+                AddContextProperties(eventData, item.Context);
+
+                // TODO: also add values from ISupportMetrics when that interface becomes public
+
+                currentInput.OnNext(eventData);
+            }
+            finally
+            {
+                this.next.Process(item);
+            }
+        }
+
+        private void AddContextProperties(EventData eventData, TelemetryContext context)
+        {
+            if (context.Component != null && !string.IsNullOrEmpty(context.Component.Version))
+            {
+                AddPayloadProperty(eventData, "context_component_version", context.Component.Version);
```

Comment:
Consider using the same tag names as Application Insights uses. `context_component_version` will become an `ai.component.version` or something like this